login/login-tool/tool/src/java/org/sakaiproject/login/config/Pac4jConfig.java

```diff
@@ -0,0 +1,137 @@
+/**********************************************************************************
+ * $URL$
+ * $Id$
+ ***********************************************************************************
+ *
+ * Copyright (c) 2005, 2006, 2008 The Sakai Foundation
+ *
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **********************************************************************************/
+
+package org.sakaiproject.login.config;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.pac4j.cas.client.CasClient;
+import org.pac4j.cas.config.CasConfiguration;
+import org.pac4j.core.client.Clients;
+import org.pac4j.core.config.Config;
+import org.pac4j.saml.client.SAML2Client;
+import org.pac4j.saml.config.SAML2Configuration;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * PAC4J configuration for CAS and SAML authentication protocols.
+ * Replaces the legacy spring-security-saml2-core and jasig.cas.client implementations.
+ */
+@Slf4j
+@Configuration
+public class Pac4jConfig {
+
+    @Value("${sakai.login.cas.server.login.url:}")
+    private String casServerLoginUrl;
+
+    @Value("${sakai.login.cas.server.prefix.url:}")
+    private String casServerPrefixUrl;
+
+    @Value("${sakai.login.saml.keystore.path:}")
+    private String samlKeystorePath;
+
+    @Value("${sakai.login.saml.keystore.password:}")
+    private String samlKeystorePassword;
+
+    @Value("${sakai.login.saml.sp.metadata.path:}")
+    private String samlSpMetadataPath;
+
+    @Value("${sakai.login.saml.idp.metadata.path:}")
+    private String samlIdpMetadataPath;
+
+    @Value("${sakai.login.saml.entity.id:SakaiSAMLApp}")
+    private String samlEntityId;
```

Comment:
should come from ServerConfigurationService

Reply:
yeah i agree. what is this form of configuration? spring style?

Reply:
up to you could be simple SCS config